flink-kubernetes-operator-autoscaler/src/test/java/org/apache/flink/kubernetes/operator/autoscaler/BacklogBasedScalingTest.java

```diff
@@ -1,437 +0,0 @@
-        var now = Instant.ofEpochMilli(0);
-        setClocksTo(now);
-        app.getStatus().getJobStatus().setUpdateTime(String.valueOf(now.toEpochMilli()));
-        metricsCollector.setCurrentMetrics(
-                Map.of(
-                        source1,
-                        Map.of(
-                                FlinkMetric.NUM_RECORDS_OUT_PER_SEC,
-                                new AggregatedMetric("", Double.NaN, Double.NaN, Double.NaN, 500.),
-                                FlinkMetric.NUM_RECORDS_IN_PER_SEC,
-                                new AggregatedMetric("", Double.NaN, Double.NaN, Double.NaN, 500.)),
-                        sink,
-                        Map.of(
-                                FlinkMetric.NUM_RECORDS_IN_PER_SEC,
-                                new AggregatedMetric(
-                                        "", Double.NaN, Double.NaN, Double.NaN, 500.))));
-
-        autoscaler.scale(getResourceContext(app, ctx));
-        assertFalse(AutoScalerInfo.forResource(app, kubernetesClient).getMetricHistory().isEmpty());
-    }
-
-    @Test
-    public void testEventOnError() {
-        // Invalid config
-        app.getSpec()
-                .getFlinkConfiguration()
-                .put("kubernetes.operator.job.autoscaler.enabled", "3");
-        autoscaler.scale(getResourceContext(app, createAutoscalerTestContext()));
-
-        var event = eventCollector.events.poll();
-        assertTrue(eventCollector.events.isEmpty());
-        assertEquals(EventRecorder.Reason.AutoscalerError.toString(), event.getReason());
-        assertTrue(event.getMessage().startsWith("Could not parse"));
-    }
-
-    private void redeployJob(Instant now) {
-        app.getStatus().getJobStatus().setUpdateTime(String.valueOf(now.toEpochMilli()));
-    }
-
-    private void setClocksTo(Instant time) {
-        var clock = Clock.fixed(time, ZoneId.systemDefault());
-        metricsCollector.setClock(clock);
-        scalingExecutor.setClock(clock);
-    }
-
-    @NotNull
-    private TestUtils.TestingContext<HasMetadata> createAutoscalerTestContext() {
-        return new TestUtils.TestingContext<>() {
-            public <T1> Set<T1> getSecondaryResources(Class<T1> aClass) {
-                return (Set)
-                        kubernetesClient.configMaps().inAnyNamespace().list().getItems().stream()
-                                .collect(Collectors.toSet());
-            }
-        };
-    }
-
-    private void assertFlinkMetricsCount(
-            int scalingCount, int balancedCount, TestUtils.TestingContext<HasMetadata> ctx) {
-        AutoscalerFlinkMetrics autoscalerFlinkMetrics =
-                autoscaler.flinkMetrics.get(
-                        ResourceID.fromResource(getResourceContext(app, ctx).getResource()));
-        assertEquals(scalingCount, autoscalerFlinkMetrics.numScalings.getCount());
-        assertEquals(balancedCount, autoscalerFlinkMetrics.numBalanced.getCount());
-    }
-}
```

Comment:
Can we keep this test class and keep the refactoring to a minimum in this feature PR?